x-pack/plugins/ml/public/components/job_selector/job_selector.js

```diff
@@ -0,0 +1,457 @@
+            })}
+          </EuiText>
+        </EuiLink>);
+    } else if (showAllBadges === true && newSelection.length > BADGE_LIMIT) {
+      badges.push(
+        <EuiLink
+          key="hide-badges-link"
+          onClick={() => setShowAllBadges(!showAllBadges)}
+        >
+          <EuiText grow={false} size="xs">
+            {i18n.translate('xpack.ml.jobSelector.hideFlyoutBadges', {
+              defaultMessage: 'Hide'
+            })}
+          </EuiText>
+        </EuiLink>);
+    }
+
+    return badges;
+  }
+
+  function renderJobSelectionBar() {
+    return (
+      <EuiFlexGroup responsive={false} gutterSize="xs" alignItems="center">
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            onClick={handleJobSelectionClick}
+          >
+            {i18n.translate('xpack.ml.jobSelector.jobSelectionButton', {
+              defaultMessage: 'Job Selection'
+            })}
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup wrap responsive={false} gutterSize="xs" alignItems="center">
+            {renderIdBadges()}
+          </EuiFlexGroup>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  function renderFlyout() {
+    if (isFlyoutVisible) {
+      return (
+        <EuiFlyout
+          onClose={closeFlyout}
+          aria-labelledby="Job Selection"
+          size="l"
+        >
+          <EuiFlyoutHeader
+            hasBorder
+            className="mlJobSelectorFlyoutHeader"
+          >
+            <EuiTitle size="m">
+              <h2 id="flyoutTitle">
+                {i18n.translate('xpack.ml.jobSelector.flyoutTitle', {
+                  defaultMessage: 'Job Selection'
+                })}
+              </h2>
+            </EuiTitle>
+          </EuiFlyoutHeader>
+          <EuiFlyoutBody className="mlJobSelectorFlyoutBody">
+            <EuiFlexGroup direction="column" responsive={false}>
+              <EuiFlexItem grow={false}>
+                <EuiFlexGroup wrap responsive={false} gutterSize="xs" alignItems="center">
+                  {renderNewSelectionIdBadges()}
+                </EuiFlexGroup>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiFlexGroup direction="row" justifyContent="spaceBetween" responsive={false}>
+                  <EuiFlexItem grow={false}>
+                    {!singleSelection &&
+                    <EuiButtonEmpty
+                      onClick={clearSelection}
+                      size="xs"
+                    >
+                      {i18n.translate('xpack.ml.jobSelector.clearAllFlyoutButton', {
+                        defaultMessage: 'Clear All'
+                      })}
+                    </EuiButtonEmpty>}
+                  </EuiFlexItem>
+                  <EuiFlexItem grow={false}>
+                    <EuiSwitch
+                      label="Apply timerange"
```

Comment:
Should be `Apply time range`. Is it possible to add some info icon to clarify what this control does?